repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/reflection/ReflectionTools.java

```diff
@@ -38,24 +38,7 @@ private ReflectionTools() {
         for (Constructor<T> constructor : (Constructor<T>[]) clazz.getConstructors()) {
             final Class<?>[] constructorParameters = constructor.getParameterTypes();
 
-            if (parameters.length != constructorParameters.length) {
-                continue;
-            }
-
-            boolean suitable = true;
-
-            for (int i = 0; i < parameters.length; i++) {
-                if (parameters[i] == null) {
-                    continue;
-                }
-
-                if (!constructorParameters[i].isAssignableFrom(parameters[i])) {
-                    suitable = false;
-                    break;
-                }
-            }
-
-            if (suitable) {
+            if (parametersMatch(constructorParameters, parameters)) {
```

Comment:
Seems like a waste to create the `constructorParameters` variable now.

You could also replace the method with:
```
return Arrays.stream((Constructor<T>[]) clazz.getConstructors())
        .filter(constructor -> parametersMatch(constructor.getParameterTypes(), parameters))
        .findFirst()
        .orElseThrow(() -> new NoSuchMethodException("No constructor found with expected signature for class: " + clazz.getCanonicalName()));
```